### PR TITLE
Implemented Abort Signup (Go to Login)

### PR DESCRIPTION
**Pull request by bryjjch**
description: Implemented Abort Signup (Go to Login)

---

- [x] only docs were changed, the checklist below is left unchecked as it is not relevant

checklist:

- [ ] coding style (conforms to Google Java Style Guide) 
- [ ] especially follows section 7 (Javadoc is present for every public class, public or protected member)
- [ ] especially follows section 4.6 (Multiple consecutive blank lines are not encouraged).
- [ ] correctness (follows CA and SOLID, is a reasonable and not overly complicated solution, etc.)
- [ ] tests (contains tests with reasonable coverage)
- [ ] description of pull request is clear, consise, and reasonably detailed

### CODE REVIEW

File: src/services/signup_abort/SignupAbortInteractor.java [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
 
-public class SignupAbortInteractor {
+public class SignupAbortInteractor implements SignupAbortInputBoundary{
+    private SignupAbortOutputBoundary userPresenter;
+    private SignupAbortUserDataAccessInterface userDataAccess;
+
+    public SignupAbortInteractor(
+            SignupAbortOutputBoundary outputBoundary,
+            SignupAbortUserDataAccessInterface userDataAccess
+    ) {
```
Comment (HussainE4) [inline src/services/signup_abort/SignupAbortInteractor.java:10]: Please check if this matches the Java Style Guide

File: src/services/signup_abort/SignupAbortInteractor.java [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+        this.userPresenter = outputBoundary;
+        this.userDataAccess = userDataAccess;
+    }
+
+    @Override
+    public void initiateLoginProcess() {
+        SignupAbortOutputData signupAbortOutputData = new SignupAbortOutputData();
+        userPresenter.presentLoginView(signupAbortOutputData);
```
Comment (HussainE4) [inline src/services/signup_abort/SignupAbortInteractor.java:18]: Do we even need an output data class for this use case. I saw that Output data was empty class so maybe it'd be cleaner to just remove it.

File: src/services/signup_abort/SignupAbortOutputData.java [hunk 3]
```diff
@@ -1,4 +1,7 @@
 package services.signup_abort;
 
 public class SignupAbortOutputData {
+    public SignupAbortOutputData() {
+
+    }
 }
```
Comment (HussainE4) [inline src/services/signup_abort/SignupAbortOutputData.java:7]: Instead of empty class, we could remove this

Comment (skrukwa) [reply src/services/signup_abort/SignupAbortOutputData.java:7]: yes delete the file

File: src/services/signup_abort/SignupAbortInputBoundary.java [hunk 4]
```diff
@@ -1,4 +1,5 @@
 package services.signup_abort;
 
 public interface SignupAbortInputBoundary {
+    void initiateLoginProcess();
```
Comment (skrukwa) [inline src/services/signup_abort/SignupAbortInputBoundary.java:4]: use "execute()" name to be consistent with all other use cases

**Review by skrukwa — CHANGES_REQUESTED** [review]

File: src/services/signup_abort/SignupAbortController.java [hunk 5]
```diff
@@ -1,4 +1,14 @@
...
 
 public class SignupAbortController {
+
+  final SignupAbortInputBoundary signupAbortInteractor;
+
+  public SignupAbortController(SignupAbortInputBoundary signupAbortInteractor) {
+    this.signupAbortInteractor = signupAbortInteractor;
+  }
+
+  public void goToLogin() {
```
Comment (umututku03) [inline src/services/signup_abort/SignupAbortController.java:11]: It could be good to rephrase it as prepareSuccessView()

**Review by umututku03 — APPROVED** [review]

File: src/services/signup_abort/SignupAbortController.java (hunk 5 above)
Comment (HussainE4) [reply src/services/signup_abort/SignupAbortController.java:11]: I agree as that is what we are using for the other use cases

Comment (bryjjch) [reply src/services/signup_abort/SignupAbortController.java]: Ok I changed it

**Review by skrukwa — APPROVED** [review]

**Pull request merged**